rust/crates/sift_cli/src/cmd/test_server/mod.rs

```diff
@@ -0,0 +1,101 @@
+use super::Context;
+use crate::cmd::test_server::metrics_streaming_client::MetricsStreamingClient;
+use crate::{cli::TestServerArgs, util::tty::Output};
+use anyhow::Result;
+use server::TestServer;
+use sift_rs::assets::v1::asset_service_server::AssetServiceServer;
+use sift_rs::ingest::v1::ingest_service_server::IngestServiceServer;
+use sift_rs::ingestion_configs::v2::ingestion_config_service_server::IngestionConfigServiceServer;
+use sift_rs::ping::v1::ping_service_server::PingServiceServer;
+use std::process::ExitCode;
+use std::sync::Arc;
+use tokio::sync::mpsc;
+use tokio::sync::watch;
+use tonic::transport::Server;
+use tonic_reflection::server::Builder;
+
+pub const FILE_DESCRIPTOR_SET: &[u8] = include_bytes!("../../../descriptor.bin");
+pub mod metrics_streaming_client;
+pub mod server;
+use crate::cmd::test_server::metrics_streaming_client::Metrics;
+
+pub async fn run(ctx: Context, args: TestServerArgs) -> Result<ExitCode> {
+    let local_address = args.local_address.unwrap_or("127.0.0.1:50051".into());
+    let addr = local_address.parse()?;
```

Comment:
Generally helpful to include contextual information with errors for better traceability:

```rust
use anyhow::{Result, Context};

let addr = local_address.parse().context("foobar")?;
```

Reply:
Same for all other places.